l10n_es_intrastat_product/models/l10n_es_intrastat_product_declaration.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+# Copyright 2009-2017 Noviat.
+# © 2016 - FactorLibre - Ismael Calvo <ismael.calvo@factorlibre.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from openerp import api, fields, models, _
+from openerp.exceptions import Warning as UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class L10nEsIntrastatProductDeclaration(models.Model):
+    _name = 'l10n.es.intrastat.product.declaration'
+    _description = "Intrastat Product Declaration for Spain"
+    _inherit = ['intrastat.product.declaration', 'mail.thread']
+
+    computation_line_ids = fields.One2many(
+        comodel_name='l10n.es.intrastat.product.computation.line',
+        inverse_name='parent_id',
+        string='Intrastat Product Computation Lines',
+        states={'done': [('readonly', True)]})
+    declaration_line_ids = fields.One2many(
+        comodel_name='l10n.es.intrastat.product.declaration.line',
+        inverse_name='parent_id',
+        string='Intrastat Product Declaration Lines',
+        states={'done': [('readonly', True)]})
+
+    def _get_intrastat_transaction(self, inv_line):
+        transaction = super(
+            L10nEsIntrastatProductDeclaration, self
+        )._get_intrastat_transaction(inv_line)
+        if not transaction:
+            module = __name__.split('addons.')[1].split('.')[0]
+            transaction = self.env.ref(
+                '%s.intrastat_transaction_1' % module)
+        return transaction
+
+    def _get_intrastat_state(self, inv_line):
+        """
+        Same logic as in product intrastat _get_region method.
+        """
+        intrastat_state = False
+        inv_type = inv_line.invoice_id.type
+        if inv_line.move_line_ids:
+            if inv_type in ('in_invoice', 'out_refund'):
+                intrastat_state = inv_line.move_line_ids[0].location_id.\
```

Comment:
This is a relation between `account.invoice.line` and `stock.move`. 

Are you sure this code works in a standard odoo installation?

Maybe a dependency is missing: stock_picking_invoice_link?

Reply:
This module depends on intrastat_product that depends on stock_picking_invoice_link at the same time. This should be covered.